src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -479,7 +479,15 @@ emit_xcompare_for_intrinsic (MonoCompile *cfg, MonoClass *klass, int intrinsic_i
 static MonoInst*
 emit_xequal (MonoCompile *cfg, MonoClass *klass, MonoInst *arg1, MonoInst *arg2)
 {
+#ifdef TARGET_ARM64
+	int size = mono_class_value_size (klass, NULL);
```

Comment:
Why does this need a separate opcode ? It can just fall back to the normal OP_XEQUAL code for 64 bit vectors.

Reply:
I find it more clear to have a separate opcode and check if it is 128bit vector to decide which code path it should goes to.